plume-common/src/activity_pub/inbox.rs

```diff
@@ -1,48 +1,609 @@
+                    None => return Inbox::NotHandled(ctx, act, InboxError::InvalidID),
+                };
+
+                // Get the actor ID
+                let actor_id = match get_id(act["actor"].clone()) {
+                    Some(x) => x,
+                    None => return Inbox::NotHandled(ctx, act, InboxError::InvalidActor(None)),
+                };
+                // Transform this actor to a model (see FromId for details about the from_id function)
+                let actor = match A::from_id(
+                    ctx,
+                    &actor_id,
+                    serde_json::from_value(act["actor"].clone()).ok(),
+                ) {
+                    Ok(a) => a,
+                    // If the actor was not found, go to the next handler
+                    Err((json, e)) => {
+                        if let Some(json) = json {
+                            act["actor"] = json;
+                        }
+                        return Inbox::NotHandled(ctx, act, InboxError::InvalidActor(Some(e)));
+                    }
+                };
 
-    fn notify(&self, conn: &C) -> Result<(), Self::Error>;
+                // Same logic for "object"
+                let obj_id = match get_id(act["object"].clone()) {
+                    Some(x) => x,
+                    None => return Inbox::NotHandled(ctx, act, InboxError::InvalidObject(None)),
+                };
+                let obj = match M::from_id(
+                    ctx,
+                    &obj_id,
+                    serde_json::from_value(act["object"].clone()).ok(),
+                ) {
+                    Ok(o) => o,
+                    Err((json, e)) => {
+                        if let Some(json) = json {
+                            act["object"] = json;
+                        }
+                        return Inbox::NotHandled(ctx, act, InboxError::InvalidObject(Some(e)));
+                    }
+                };
+
+                // Handle the activity
+                match obj.activity(ctx, actor, &act_id) {
+                    Ok(res) => Inbox::Handled(res.into()),
+                    Err(e) => Inbox::Failed(e),
+                }
+            } else {
+                // If the Activity type is not matching the expected one for
+                // this handler, try with the next one.
+                Inbox::NotHandled(ctx, act, e)
+            }
+        } else {
+            self
+        }
+    }
+
+    /// Transforms the inbox in a `Result`
+    pub fn done(self) -> Result<R, E> {
+        match self {
+            Inbox::Handled(res) => Ok(res),
+            Inbox::NotHandled(_, _, err) => Err(E::from(err)),
+            Inbox::Failed(err) => Err(err),
+        }
+    }
 }
 
-pub trait Deletable<C, A> {
-    type Error;
+/// Get the ActivityPub ID of a JSON value.
+///
+/// If the value is a string, its value is returned.
+/// Otherwise we return its `id` field, if it is a string
+///
+/// # Panics
```

Comment:
I don't think it can actually panic? Also the `Otherwise we return its 'id' field, if it is a string` is a bit unclear

Reply:
Yeah, I think I forgot to update the docs.